airflow/ui/src/pages/Pools/PoolBar.tsx

```diff
@@ -0,0 +1,109 @@
+/*!
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import { Box, Flex, Text, VStack } from "@chakra-ui/react";
+import { MdHourglassDisabled, MdHourglassFull } from "react-icons/md";
+
+import type { PoolResponse } from "openapi/requests/types.gen";
+import { Tooltip } from "src/components/ui";
+import { stateColor } from "src/utils/stateColor";
+
+type PoolBarProps = {
+  readonly pools: Array<PoolResponse>;
+};
+
+const PoolBar = ({ pools }: PoolBarProps) => (
+  <Flex direction="column" gap={4}>
+    {pools.map((pool) => {
```

Comment:
I think this part should be in `Pools.tsx` and this component should be for only a single Pool.